File: leetcode/two-sum/javascript/two-sum-one-pass-hash-table.js
const twoSum = function(numbers, target) {
    const map = new Map();
    for (let i = 0, l = numbers.length; i < l; i += 1) {
        const complement = target - numbers[i];
        if (map.has(complement) && map.get(complement) != i) {
            return [map.get(complement), i];
        }
        map.set(numbers[i], i);
    }
    return [];
};

console.log(twoSum([2, 4, 5, 6, 11], 10));

File: leetcode/two-sum/python/two_sum_one_pass_hash_table.py
class Solution:
    def two_sum(self, nums:list[int], target:int) -> list[int]:
        hashmap = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in hashmap:
                return [hashmap[complement], i]
            hashmap[nums[i]] = i
        return []

sol = Solution()
print(sol.two_sum([1, 3, 4, 6, 7, 11], 9))